examples/rbfe_single.py

```diff
@@ -15,11 +16,89 @@
 from testsystems.relative import hif2a_ligand_pair
 
 from ff.handlers.serialize import serialize_handlers
-from ff.handlers import nonbonded
+from ff.handlers.nonbonded import AM1CCCHandler, LennardJonesHandler
 from parallel.client import CUDAPoolClient
 
+from typing import Union, Optional, Iterable, Any, Tuple, Dict
 
-def wrap_method(args, fxn):
+from scipy.optimize import root_scalar
+
+array = Union[np.array, jnp.array]
+Handler = Union[AM1CCCHandler, LennardJonesHandler] # TODO: do these all inherit from a Handler class already?
+
+
+# TODO: move optimizer stuff into timemachine/optimize or so
+def _taylor_first_order(x: array, f_x: float, grad: array) -> callable:
+    """
+
+    Notes:
+        TODO: is it preferable to use jax linearize? https://jax.readthedocs.io/en/latest/jax.html#jax.linearize
+    """
+    def f_prime(y: array) -> float:
+        return f_x + np.dot(grad, y - x)
+
+    return f_prime
+
+
+def _smart_clip(
+        x: array, f_x: float, grad: array,
+        step_size: float=0.1, search_direction: Optional[array]=None,
```

Comment:
When should we use the default values for step_size and step_lower_bound?

Reply:
Will add some usage notes. `step_size` is used to generate an initial proposal `x_proposed`. If `f_prime(x_proposed) < step_lower_bound`, then the step will be truncated.

The default `step_lower_bound=0` corresponds to a suggestion in the cited study, incorporating the knowledge that the loss is bounded below by 0. In the script, I passed in a non-default argument to the `step_lower_bound` to make the behavior of the method more conservative.

The default value `step_size=0.1` isn't very precisely chosen. The behavior of the method will be insensitive to picking `step_size` anywhere between like 1e-3 and +inf for our problems, since this will trigger the step-truncating logic on most every step. If the `step_size` is chosen sufficiently small that it rarely produces proposals that violate `step_lower_bound`, then that will start to have an effect on the behavior of the optimizer.

Reply:
Thanks - that description definitely helped me out!

Reply:
Added these clarifications to docstring in https://github.com/proteneer/timemachine/pull/346/commits/d3a31d2e60b7732ab8b254094cefa2cb67835093